pandas/_libs/tslibs/strptime.pyx

```diff
@@ -0,0 +1,631 @@
+# -*- coding: utf-8 -*-
+# cython: profile=False
+"""Strptime-related classes and functions.
+
+CLASSES:
+    LocaleTime -- Discovers and stores locale-specific time information
+    TimeRE -- Creates regexes for pattern matching a string of text containing
+                time information
+
+FUNCTIONS:
+    _getlang -- Figure out what language is being used for the locale
+    strptime -- Calculates the time struct represented by the passed-in string
+
+"""
+import time
+import locale
+import calendar
+import re
+
+
+# Python 2 vs Python 3
+try:
+    from thread import allocate_lock as _thread_allocate_lock
+except:
+    try:
+        from _thread import allocate_lock as _thread_allocate_lock
+    except:
+        try:
+            from dummy_thread import allocate_lock as _thread_allocate_lock
+        except:
+            from _dummy_thread import allocate_lock as _thread_allocate_lock
+
+
+from cython cimport Py_ssize_t
+from cpython cimport PyFloat_Check
+
+cimport cython
+
+import numpy as np
+cimport numpy as np
+from numpy cimport ndarray, int64_t
+
+from datetime import date as datetime_date
+from datetime cimport datetime
+
+# This is src/datetime.pxd
+from datetime cimport (
+    PANDAS_FR_ns,
+    check_dts_bounds,
+    pandas_datetimestruct,
+    pandas_datetimestruct_to_datetime)
+
+cimport util
+from util cimport is_string_object, get_nat
+
+cdef int64_t NPY_NAT = util.get_nat()
+
+cdef set _nat_strings = set(['NaT', 'nat', 'NAT', 'nan', 'NaN', 'NAN'])
+
+
```

Comment:
can we move both of these to util.pxd? to avoid repeating code

Reply:
`_nat_strings` and what else?

Reply:
checknull_with_nat, this seems like its repeating in lots of places.